backend/src/db/migrations/20240318164718_dynamic-secret.ts

```diff
@@ -0,0 +1,58 @@
+import { Knex } from "knex";
+
+import { SecretEncryptionAlgo, SecretKeyEncoding, TableName } from "../schemas";
+import { createOnUpdateTrigger, dropOnUpdateTrigger } from "../utils";
+
+export async function up(knex: Knex): Promise<void> {
+  const doesTableExist = await knex.schema.hasTable(TableName.DynamicSecret);
+  if (!doesTableExist) {
+    await knex.schema.createTable(TableName.DynamicSecret, (t) => {
+      t.uuid("id", { primaryKey: true }).defaultTo(knex.fn.uuid());
+      t.string("slug").notNullable();
```

Comment:
I noticed this is just the dynamic secret name and not the actual slug. Instead of having a separate slug field, can you just make sure this name follows `max 64 characters and alphanumeric with '+=,.@-_' characters` like aws? This way we don't need to have separate field just for easy look up

Reply:
Also i would just rename this to name